test/allowlist_for_publicAPI.json

```diff
@@ -514,6 +514,9 @@
     "Tuple",
     "reduce"
   ],
+  "torch.ao.sparsity.sparsifier.nearly_diagonal_sparsifier": [
```

Comment:
You should not add new entries to this file.
Make sure to follow the guidelines from the error message to make this properly public or private depending on what it should be.